lib/spack/spack/spec.py

```diff
@@ -1546,41 +1545,41 @@ def package_hash(self):
     def dag_hash(self, length=None):
         """This is Spack's default hash, used to identify installations.
 
-        At the moment, it excludes build dependencies to avoid rebuilding
-        packages whenever build dependency versions change. We will
-        revise this to include more detailed provenance when the
-        concretizer can more aggressievly reuse installed dependencies.
+        This is the finest-grained hash, and includes build dependencies as
+        well as the package hash, so that we know when package files have
+        changed between builds.  Previously, the DAG hash excluded build
+        dependencies and the package hash, and the full_hash provided that
+        functionality.  But the full_hash was deprecated and the dag_hash
+        modified to behave in the same way so that install trees, the db,
+        environments, and buildcaches would all be keyed by the finest
+        grained hash.
```

Comment:
I think we can leave this level of detail to the commit comments and remove everything after "previously" here.